interview-questions.md

```diff
@@ -4,49 +4,53 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. Why would you use the method super()?
 
-  Your answer:
+  Your answer: Super is used in a constructor, required by React
 
-  Researched answer:
+  Researched answer: Super() will calls the constructor of its parent class. This is required when you need to access some variables from the parent class. In React, when you call super with props. React will make props available across the component through this.props .
 
 
 
 2. What is a virtual DOM?
 
-  Your answer:
+  Your answer: A part of react, i cant remember exactly
 
-  Researched answer:
+  Researched answer: The virtual DOM (VDOM) is a programming concept where an ideal, or “virtual”, representation of a UI is kept in memory and synced with the “real” DOM by a library such as ReactDOM. This process is called reconciliation.
 
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
-  Your answer:
+  Your answer: I'm not sure.
 
   Researched answer:
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JSX is a coding language used in React that is a hybrid of JavaScript and html
+
+  Researched answer: JSX is a syntax extension to JavaScript. It's used with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with the full power of JavaScript.
+
 
-  Researched answer:
 
 
 
 5. What is state in React?
 
-  Your answer:
+  Your answer: State is an component used in React to help render specified outcomes
```

Comment:
Not quite, its not a component. Its an object and it helps store data/information.